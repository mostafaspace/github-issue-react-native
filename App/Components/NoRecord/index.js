import React from 'react';
import {Text, View} from 'react-native';
import Styles from './styles';
import PropTypes from 'prop-types';

export function NoRecord({containerStyle, text}) {
    return (
        <View style={[Styles.emptyListContainer, containerStyle]} testID={'NoRecord'}>
            <Text style={Styles.emptyListText}>{text}</Text>
        </View>
    );
}

NoRecord.propTypes = {
    containerStyle: PropTypes.object,
    text: PropTypes.string,
};
