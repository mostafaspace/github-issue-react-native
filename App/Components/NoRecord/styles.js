import {StyleSheet} from 'react-native';

export default StyleSheet.create({
    emptyListContainer: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
    },
    emptyListText: {
        lineHeight: 18,
        color: '#222',
        textAlign: 'center',
        margin: 16,
    },
});
