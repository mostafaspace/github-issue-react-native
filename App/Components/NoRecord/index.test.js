import React from 'react';
import {render, cleanup} from '@testing-library/react-native';
import {NoRecord} from '../NoRecord';
import renderer from 'react-test-renderer';
import '@testing-library/jest-native/extend-expect';

afterEach(cleanup);
it('renders component', () => {
    render(<NoRecord text={'test'} />);
});

it('check if has text', () => {
    const {queryByTestId, getByText} = render(<NoRecord text={'text'} />);
    expect(queryByTestId('NoRecord')).toContainElement(getByText('text'));
});

it('NoRecord matches snapshot', () => {
    const tree = renderer.create(<NoRecord text={'test'} />).toJSON();
    expect(tree).toMatchSnapshot();
});
