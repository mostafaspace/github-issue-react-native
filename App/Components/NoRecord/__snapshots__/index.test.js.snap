// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoRecord matches snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      },
      undefined,
    ]
  }
  testID="NoRecord"
>
  <Text
    style={
      Object {
        "color": "#222",
        "lineHeight": 18,
        "margin": 16,
        "textAlign": "center",
      }
    }
  >
    test
  </Text>
</View>
`;
